### PR TITLE
fix: remove extra return in machine template

### PR DESCRIPTION
**Pull request by matttrach**
addresses https://github.com/rancher/terraform-provider-rancher2/issues/1398

### CODE REVIEW

**Comment by matttrach** [comment]
When accepted, this will need to be backported to v5 and v4

File: rancher2/resource_rancher2_machine_config_v2.go [hunk 1]
```diff
@@ -60,10 +60,6 @@ func waitForMachineConfigV2(d *schema.ResourceData, config *Config, interval tim
 			return nil
 		}
 		log.Printf("[INFO] Retrying on error Refreshing Machine Config V2 %s: %v", d.Id(), err)
-		if IsNotFound(err) || IsForbidden(err) {
```
Comment (jiaqiluo) [inline rancher2/resource_rancher2_machine_config_v2.go:63]: I don't believe the for loop is being used correctly in this function. If getMachineConfigV2ByID returns an error, the function will remain stuck in the loop indefinitely and never reach the select block. I’d recommend refactoring the logic to use a backoff function, similar to [OnError](https://pkg.go.dev/k8s.io/client-go/util/retry@v0.30.1#OnError), if you prefer not to introduce a new module.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: rancher2/resource_rancher2_machine_config_v2.go (hunk 1 above)
Comment (matttrach) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: but the select block is inside the loop

Comment (matttrach) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: with this change there are only two return scenarios:
1. getMachineConfigV2ByID returns nil for the error (there wasn't an error)
2. the time is after the timeout and the context is done

am I missing something?

Comment (matttrach) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: I am not a fan of loops that don't end like this one, but without a good test suite (something I am working on) I am trying to reduce impact as much as possible.

Comment (jiaqiluo) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: My bad. I misread the code and thought the select part was outside of the for loop when reviewing the code.  
~But we still have the problem that the loop never ends: if `getMachineConfigV2ByID` somehow always returns a 404/403 error, then the function goes to the `case <-time.After(rancher2RetriesWait * time.Second):` case and keep retrying.  Right?~

Never mind, the `ctx` will time out and make the function exit the loop.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: rancher2/resource_rancher2_machine_config_v2.go (hunk 1 above)
Comment (jiaqiluo) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: I think we should unset the ID in the case of timing out, ie. adding `d.SetId("")` to the  `case <-ctx.Done():` section.

Comment (matttrach) [reply rancher2/resource_rancher2_machine_config_v2.go:63]: done, thank you so much for your help!

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**